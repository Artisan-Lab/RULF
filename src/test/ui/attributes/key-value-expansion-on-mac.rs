#![feature(rustc_attrs)]

#[rustc_dummy = stringify!(a)] // OK
macro_rules! bar {
    () => {};
}

// FIXME?: `bar` here expands before `stringify` has a chance to expand.
// `#[rustc_dummy = ...]` is validated and dropped during expansion of `bar`,
// the "unexpected expression" errors comes from the validation.
#[rustc_dummy = stringify!(b)] //~ ERROR unexpected expression: `stringify!(b)`
bar!();

fn main() {}
