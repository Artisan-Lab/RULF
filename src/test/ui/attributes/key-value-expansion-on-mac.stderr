error: unexpected expression: `stringify!(b)`
  --> $DIR/key-value-expansion-on-mac.rs:11:17
   |
LL | #[rustc_dummy = stringify!(b)]
   |                 ^^^^^^^^^^^^^

error: aborting due to previous error

